lib/packwerk.rb

```diff
@@ -8,81 +8,9 @@
 # Provides String#pluralize
 require "active_support/core_ext/string"
 
-module Packwerk
-  extend ActiveSupport::Autoload
-
-  autoload :ApplicationLoadPaths
-  autoload :ApplicationValidator
-  autoload :AssociationInspector
-  autoload :OffenseCollection
-  autoload :Cache
-  autoload :Cli
-  autoload :Configuration
-  autoload :ConstantDiscovery
-  autoload :ConstantNameInspector
-  autoload :ConstNodeInspector
-  autoload :DeprecatedReferences
-  autoload :FileProcessor
-  autoload :FilesForProcessing
-  autoload :Graph
-  autoload :Node
-  autoload :NodeHelpers
-  autoload :NodeProcessor
-  autoload :NodeProcessorFactory
-  autoload :NodeVisitor
-  autoload :Offense
-  autoload :OffensesFormatter
-  autoload :OutputStyle
-  autoload :Package
-  autoload :PackageSet
-  autoload :ParsedConstantDefinitions
-  autoload :Parsers
-  autoload :ParseRun
-  autoload :UnresolvedReference
-  autoload :Reference
-  autoload :ReferenceExtractor
-  autoload :ReferenceOffense
-  autoload :Result
-  autoload :RunContext
-  autoload :Version
-  autoload :ViolationType
-
-  module OutputStyles
-    extend ActiveSupport::Autoload
-
-    autoload :Coloured
-    autoload :Plain
-  end
-
-  autoload_under "commands" do
-    autoload :OffenseProgressMarker
-  end
-
-  module Formatters
-    extend ActiveSupport::Autoload
-
-    autoload :OffensesFormatter
-    autoload :ProgressFormatter
-  end
+require "zeitwerk"
+loader = Zeitwerk::Loader.for_gem
```

Comment:
Usually the Zeitwerk loader is exposed via a reader on the root module (eg. https://github.com/dry-rb/dry-monitor/blob/195c1324b7d3edfd00bd9eed8c61fb4ccc14af5f/lib/dry/monitor.rb#L23) but since this runs statically we probably don't need this unless we want to support plugin gems in the future.

Reply:
Oh interesting! Yeah, we can always make that change if that ever becomes a capability we wanted to support 👍